app/src/main/java/com/example/schmotify/LoginActivity.java

```diff
@@ -29,33 +79,199 @@ protected void onCreate(Bundle savedInstanceState) {
+                    assert response.body() != null;
+                    try {
+                        tokenResponse.postValue(new JSONObject(response.body().string()));
+                    } catch (JSONException | IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            });
+            tokenResponse.observe(LoginActivity.this, codeObserver);
+        }
+    }
+
+    final Observer<JSONObject> codeObserver = new Observer<JSONObject>() {
+        @Override
+        public void onChanged(JSONObject jsonObject) {
+            try {
+                Log.d("LOGIN INFO", jsonObject.toString());
+                mRefreshToken = jsonObject.getString("refresh_token");
+            } catch (JSONException e) {
+                throw new RuntimeException(e);
+            }
+            try {
+                mAccessToken.postValue(jsonObject.getString("access_token"));
+            } catch (JSONException e) {
+                throw new RuntimeException(e);
             }
-        });
+        }
+    };
+
+    final Observer<String> tokenObserver = new Observer<String>() {
+        @Override
+        public void onChanged(@Nullable final String s) {
+            final Request request = new Request.Builder()
+                    .url("https://api.spotify.com/v1/me")
+                    .addHeader("Authorization", "Bearer " + mAccessToken.getValue())
+                    .build();
+
+            cancelCall();
+            mCall = mOkHttpClient.newCall(request);
+
+            mCall.enqueue(new Callback() {
+                @Override
+                public void onFailure(@NonNull Call call, @NonNull IOException e) {
+                    Log.d("HTTP", "Failed to fetch data: " + e);
+                    Toast.makeText(LoginActivity.this, "Failed to fetch data, watch Logcat for more details",
+                            Toast.LENGTH_SHORT).show();
+                }
+
+                @Override
+                public void onResponse(@NonNull Call call, @NonNull Response response) throws IOException {
+                    try {
+                        assert response.body() != null;
+                        profileInfo.postValue(new JSONObject(response.body().string()));
+                    } catch (JSONException e) {
+                        Log.d("JSON", "Failed to parse data: " + e);
+                        Toast.makeText(LoginActivity.this, "Failed to parse data, watch Logcat for more details",
+                                Toast.LENGTH_SHORT).show();
+                    }
+                }
+            });
+        }
+    };
+
+    /**
+     * Get user profile
+     * This method will get the user profile using the token
+     */
+    public void getUserProfile() {
+        getCode();
+        mAccessToken.observe(this, tokenObserver);
+
+    }
+
+    //TODO (James): Update the scopes
```

Comment:
There's a TODO comment here. The scopes need to be updated so we get a token that's useful to us. The scopes we will definitely need are as follows: "user-follow-read", "user-read-email", "user-library-read", "user-top-read"

Keep in mind we will probably need more, but this should be fine for the moment